scale/scheduler/scheduling/manager.py

```diff
@@ -418,6 +421,9 @@ def _process_queue(self, nodes, job_types, job_type_limits, job_type_resources,
                 if job_type_id in job_type_limits:
                     job_type_limits[job_type_id] -= 1
 
+            if len(scheduled_job_executions) == QUEUE_LIMIT:
+                break
```

Comment:
2 questions here:

- Is it possible to pack more than `QUEUE_LIMIT` in one pass? If so, we should change to `>=`
- I'd add a logger.debug statement here to indicate that queue limit is exceeded as long as we know it won't spam 1000 of those messages a second. :-)

Reply:
I believe it adds jobs one at a time, but I'm not opposed to changing it to `>=`.